### DISCUSSION TITLE
[Enhancement] Replace some APIs with existing .NET ones

### DISCUSSION THREAD

**Discussion by ChaseFlorell** [dotnet/maui#110] [General] [answered]
## Summary
Forms duplicated a lot of APIs due to limitations at the time. Now that more and more .NET is becoming available to all platforms, now is the time to drop those duplicate API's in favor of exising ones.

## API Changes

System.Maui.Color => System.Drawing.Color
System.Maui.Point => System.Drawing.Point
.. and others

// perhaps
System.Maui.Thickness => System.Windows.Thickness (or maybe they both move to a common one)

## Intended Use Case
Reduce code maintenance within the Maui namespace and not duplicate API's unnecessarily.

**Comment by dotMorten** [comment]
Also System.Maui.Size -> System.Drawing.Size

**Comment by PureWeen** [comment, marked as answer]
That's the plan!!

**Comment by charlesroddie** [comment]
Replacing Xamarin.Forms.Color with System.Drawing.Color would remove a lot of unnecessary duplication of code and conversions between two essentially identical types (8bpc).

However the suggestion needs to be updated. There are plans to make MAUI depend on the 2D primitive graphics layer https://github.com/dotnet/Microsoft.Maui.Graphics, which is trememdous. This means that the Color type in MAUI will currently be Microsoft.Maui.Graphics.Color (16bpc). Assuming this is intentional and desirable, System.Drawing.Graphics is no longer a relevant proposal.

### Updated proposal for the Color type:

The MAUI team and future users should push for [adding a Color type to dotnet/runtime](https://github.com/dotnet/runtime/issues/32418). There are two proposals there, an improved 8bpc Color type (minus some bad design choices of System.Drawing.Color) and a ~~16bpc one~~ 32bpc one which should be 16bpc.

Microsoft.Maui and Microsoft.Maui.Graphics can then depend on one of these types. Presumably the 16bpc one to preserve the 16bpc of Microsoft.Maui.Graphics.
